Controller UI rework

cvzbynek opened the pull request:
# Change Layout of Controller

The message panel is now placed below the app tree and FSM control, as it is less important. The diagram and control buttons are resized to be uniform. The app tree's hierarchy display is enhanced with hover functionality, and the app tree detail overview features an improved layout for better usability.

# Description

_This change reorganizes the Controller layout to prioritize user interaction with the app tree and FSM control over the message panel. The hierarchy display is more intuitive with hover effects, and the app tree detail layout is optimized for clarity. Control buttons and diagrams are now equally sized for consistency._

Fixes #301 

## Type of change

- [ ] Documentation (non-breaking change that adds or improves the documentation)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Optimization (non-breaking, back-end change that speeds up or improves usability)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (requires documentation update or affects compatibility)

## Key checklist

- [x] All tests pass (e.g., `python -m pytest`)
- [x] The documentation builds and looks OK (e.g., `python -m sphinx -b html docs docs/build`)
- [x] Pre-commit hooks run successfully (e.g., `pre-commit run --all-files`)

## Further checks

- [x] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added or an issue has been opened to address this in the future. (Indicate issue here: # (issue))

dalonsoa commented on controller/templates/controller/index.html:
```diff
@@ -30,41 +184,62 @@
   <script type="module"
           src="https://cdn.jsdelivr.net/npm/@shoelace-style/shoelace@2.18.0/cdn/shoelace-autoloader.js"></script>
+  <script src="https://cdn.jsdelivr.net/npm/mermaid@11.4.1/dist/mermaid.min.js"></script>
+  <script>
+    // Initialize mermaid diagrams.
+    mermaid.initialize({
+        themeVariables: {
+            background: "white",
+            primaryColor: "#007bff",
+            nodeBorder: "#dddddd",
+            edgeLabelBackground: "transparent",
+        },
+    });
+
+    // Refresh mermaid diagrams after htmx swap.
+    document.body.addEventListener("htmx:afterSwap", (event) => {
+        const target = event.detail.target;
+        if (target && target.querySelector(".mermaid")) {
+            mermaid.run({ nodes: target.querySelectorAll(".mermaid") });
+        }
+    });
+  </script>
```
Why do we need this bit also here and not just in the partial `state_machine.html` file?

cvzbynek replied:
My bad, copypasta remain. Will disappear!

cvzbynek replied:
I would rather wait when @jamesturner246 's PR is merged into `main`, but if both of you prefer merging into `fsm_diagram`, I can do that. Result will be the same I guess.
We can discuss the colour structure after that merging in new PR (not a draft)

dalonsoa replied:
Sure, no problem. It was just to clearly separate - for review purposes - what's the contribution of this PR and what is James'. But that will only work if you branched off from James' branch. If you did it from main, lets leave it as it is.

cvzbynek replied:
I branched it from `fsm_diagram`. But still, I would wait and see how the PR ends up. And adjust accordingly.

jamesturner246 replied:
We discussed this the other day, and agreed on branching off of any WIP branches to merge into main as and when each one finishes. In any case, my PR is merged now. I'm just looking this over and it can go straight into main after.

dalonsoa commented on controller/templates/controller/index.html:
I definitely think we should consider moving all the CSS to its own file(s). With so much customisation, these HTML files are starting to look pretty messy. There's an issue open about it (https://github.com/ImperialCollegeLondon/drunc_ui/issues/214) . Probably not a priority if there are other things to do, but something we would need to tackle for the sustainability of the code.

cvzbynek replied:
Totally agree on that. When reworking layout, I usually add tons of CSS.

dalonsoa reviewed (commented):
Very nice! Visually appealing and with a useful layout.

I'm still unconvinced with the transitions being white text over a grey of green background, without border and an area around (the bg color is just behind the text). In a separate PR, we might want to reconsider the colour structure of the different bits and pieces. 

To avoid conflicts, it would be better if this branch merges into @jamesturner246 branch, `fsm_diagram` rather than into `main`.

Finally, when making changes in the UI, it will be useful to include in the PR description a screenshot or recording (if dynamic behaviour is involved).

jamesturner246 reviewed (commented):
It looks good in principle. I'll let you finish and merge in with my changes (in main) before commenting again: there's still some stragglers.

cvzbynek commented:
@jamesturner246 @dalonsoa ,

I've updated from the `main` branch and reverted to the TD version of the Mermaid chart. The layout I’m suggesting makes more sense to me, as the FSM is the main control element. With the LD version of the chart, it tends to "disappear" when I scroll down to change states (on certain resolutions).

What do you think?

![image](https://github.com/user-attachments/assets/c019a6c8-9bd0-45d8-bd03-e65943d9d9b7)

dalonsoa commented:
@cvzbynek , if it is no longer in draft, please mark it as ready to review so it makes sense for us to approve it. 

Having said that, it looks great!

jamesturner246 reviewed (approved):
@cvzbynek that's fine. I see what you mean -- it does look a little less squashed. It's got a fairly large amount of inline CSS now, which makes it less readable, but this can be addressed later.

Regarding the CSS separation, there is already an issue open for this: #214 

Given you are working a lot on the styling, I can assign this to you if you wish to take it.

cvzbynek commented:
@jamesturner246 assign me, please!
